generate_makefile.bash

```diff
@@ -206,6 +212,10 @@ KOKKOS_SETTINGS="KOKKOS_SRC_PATH=${KOKKOS_SRC_PATH}"
 
 if [ ${#COMPILER} -gt 0 ]; then
   KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
+elif
+   [ ${#COMPILER} -eq 0 ] && [ ${#KOKKOS_DEVICES} -gt 0 ]; then
```

Comment:
This doesn't check whether KOKKOS_DEVICES includes Cuda. It should only automatically change to nvcc_wrapper if COMPILER is not set and KOKKOS_DEVICES contains Cuda.

Reply:
@dalabre from our session sitting down, try something like:
```
if echo "${KOKKOS_DEVICES}" | grep -q "Cuda"; then
  COMPILER="${KOKKOS_PATH}/bin/nvcc_wrapper"                                                  
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"  
fi
```

Reply:
Testing fixes ....

Reply:
.... tweak compile setting (remove double quotes)
```
      COMPILER=${COMPDIR}/${COMPNAME}
```
and add check for Cuda device before setting Cuda compiler
```
if [ ${#COMPILER} -gt 0 ]; then
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
elif
   [ ${#COMPILER} -eq 0 ] && [[ ${KOKKOS_DEVICES} =~ .*Cuda.* ]]; then
  COMPILER="${KOKKOS_PATH}/bin/nvcc_wrapper"
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"   
fi
```
Note, this is the string-equivalent of the file-capabable *grep -q 'string' $file*; the double [[ ]]
are required for this implementation.

changes pushed to dalabre/kokkos